src/diffusers/models/unet_1d_blocks.py

```diff
@@ -453,22 +469,30 @@ def forward(self, hidden_states, temb=None):
 
 
 class AttnDownBlock1D(nn.Module):
-    def __init__(self, out_channels, in_channels, mid_channels=None):
+    def __init__(self, out_channels: int, in_channels: int, num_layers: int = 2, mid_channels: int = None):
         super().__init__()
+
+        if num_layers < 1:
+            raise ValueError("AttnDownBlock1D requires added num_layers >= 1")
+
         mid_channels = out_channels if mid_channels is None else mid_channels
 
-        self.down = Downsample1d("cubic")
+        self.down = KernelDownsample1D("cubic")
         resnets = [
             ResConvBlock(in_channels, mid_channels, mid_channels),
-            ResConvBlock(mid_channels, mid_channels, mid_channels),
-            ResConvBlock(mid_channels, mid_channels, out_channels),
         ]
         attentions = [
             SelfAttention1d(mid_channels, mid_channels // 32),
-            SelfAttention1d(mid_channels, mid_channels // 32),
-            SelfAttention1d(out_channels, out_channels // 32),
         ]
 
+        for i in range(num_layers):
+            if i < (num_layers - 1):
+                resnets.append(ResConvBlock(mid_channels, mid_channels, mid_channels))
```

Comment:
Same I don't understand why `num_layers=2` would lead to this block having 3 layers.
This is not how we do things in https://github.com/huggingface/diffusers/blob/ab1f01e63415b63937736299d3a770554c83987e/src/diffusers/models/unet_2d_blocks.py#L434 no?

Reply:
Yeah no sure how I ended up in that incorrect rabbit hole. Should've caught that one, but it has been changed as it needed to be.